src/server_manager/web_app/app.spec.ts

```diff
@@ -35,41 +32,41 @@ const TOKEN_WITH_ONE_SERVER = 'one-server-token';
 // tslint:disable-next-line:no-any
 (global as any).bringToFront = () => {};
 
+// Inject app-root element into DOM once before the test suite runs.
+beforeAll(async () => {
+  // It seems like AppRoot class is not fully loaded/initialized until the
+  // constructor, so we invoke it directly.
+  const loadAppRoot = new AppRoot();
+
+  document.body.innerHTML = "<app-root id='appRoot' language='en'></app-root>";
+});
+
 describe('App', () => {
-  it('shows intro when starting with no manual servers or DigitalOcean token', (done) => {
-    const polymerAppRoot = new FakePolymerAppRoot();
-    const app = createTestApp(polymerAppRoot, new InMemoryDigitalOceanTokenManager());
-    polymerAppRoot.events.once('screen-change', (currentScreen) => {
-      expect(currentScreen).toEqual(AppRootScreen.INTRO);
-      done();
-    });
-    app.start();
+  it('shows intro when starting with no manual servers or DigitalOcean token', async () => {
+    const appRoot = document.getElementById('appRoot') as unknown as AppRoot;
```

Comment:
Can you try `document.createElement('app-root')` instead?

I don't think you need the beforeAll initialization or to add it to the document tree.

Reply:
Oh neat, yeah, that's a much cleaner way. I will try to get that working in a follow up PR.

The two issues holding us up at the moment are:
* We access the `document` directly in a few places in the code.
  **Error:** `TypeError: Cannot read property 'setAttribute' of null`
* app-drawer-layout not working. May also be a dependency on `document`.
**Error:** `TypeError: Cannot read property 'drawerLayout' of undefined`